### PR TITLE
General menu cleanup

### PR DESCRIPTION
**Pull request by margaree**
Nothing was really needed to get dropdown working with lit menu, but there was some import cleanup/management that should be done.

### CODE REVIEW

**Comment by dlockhart** [comment]
FYI @AllanKerr as I know you're starting to look at porting this to Lit.

File: demo/dropdown-menu.html [hunk 1]
```diff
@@ -8,6 +8,7 @@
 		<script type="module" src="../../@polymer/iron-demo-helpers/demo-pages-shared-styles.js"></script>
 		<script type="module" src="../../@polymer/iron-demo-helpers/demo-snippet.js"></script>
 		<script type="module" src="../../d2l-menu/d2l-menu.js"></script>
+		<script type="module" src="../../d2l-menu/d2l-menu-item.js"></script>
```
Comment (dlockhart) [inline demo/dropdown-menu.html]: `d2l-menu.js` gets re-imported down below (I never understood why the converted did that), does it need to be consistent?

File: demo/dropdown-more.html [hunk 2]
```diff
@@ -8,6 +8,7 @@
 		<script type="module" src="../../@polymer/iron-demo-helpers/demo-pages-shared-styles.js"></script>
 		<script type="module" src="../../@polymer/iron-demo-helpers/demo-snippet.js"></script>
 		<script type="module" src="../../d2l-menu/d2l-menu.js"></script>
+		<script type="module" src="../../d2l-menu/d2l-menu-item.js"></script>
```
Comment (dlockhart) [inline demo/dropdown-more.html]: Same here -- there's a second import further down.

File: demo/dropdown-menu.html (hunk 1 above)
Comment (margaree) [reply demo/dropdown-menu.html]: ah yup probably should keep these consistent

Comment (dbatiste) [reply demo/dropdown-menu.html]: As we've done for others, we can clean up the demo by replacing a lot of the mess with the demo page component from core.  We've done that more others we converted and I think we should do the same when we convert this.  It also helps allows eliminating a lot of deps from package.json.

**Pull request merged**